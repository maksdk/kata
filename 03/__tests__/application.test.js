import fs from 'fs';
import path from 'path';
import { html } from 'js-beautify';

import run from '../src/application';

const htmlOptions = {
  preserve_newlines: false,
  unformatted: [],
};

const fixuturesPath = path.join(__dirname, '__fixtures__');
const getTree = () => html(document.body.innerHTML, htmlOptions);

beforeAll(() => {
  const initHtml = fs.readFileSync(path.join(fixuturesPath, 'index.html')).toString();
  document.documentElement.innerHTML = initHtml;
  run();
});

test('application', async () => {
  expect(getTree()).toMatchSnapshot();

  const name = document.querySelector('[data-editable-target="name"]');
  name.click();
  expect(getTree()).toMatchSnapshot();

  const form1 = document.querySelector('form');
  const input1 = form1.elements.name;
  input1.setAttribute('value', 'Hexlet');
  form1.dispatchEvent(new Event('submit'));
  expect(getTree()).toMatchSnapshot();

  const email = document.querySelector('[data-editable-target="email"]');
  email.click();
  expect(getTree()).toMatchSnapshot();

  const form2 = document.querySelector('form');
  const input2 = form2.elements.email;
  input2.setAttribute('value', 'support@hexlet.io');
  form2.dispatchEvent(new Event('submit'));
  expect(getTree()).toMatchSnapshot();

  name.click();
  const form3 = document.querySelector('form');
  const input3 = form3.elements.name;
  input3.setAttribute('value', '');
  form3.dispatchEvent(new Event('submit'));
  expect(getTree()).toMatchSnapshot();
});
