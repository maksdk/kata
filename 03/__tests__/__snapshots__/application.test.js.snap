// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`application 1`] = `
"<div class=\\"container m-3\\">
    <div data-editable-target=\\"name\\">
        <i>name</i>
    </div>
    <div data-editable-target=\\"email\\">
        <i>email</i>
    </div>
</div>"
`;

exports[`application 2`] = `
"<div class=\\"container m-3\\">
    <div data-editable-target=\\"name\\">
        <form><input type=\\"text\\" name=\\"name\\" value=\\"\\"><input type=\\"submit\\" value=\\"Save\\"></form>
    </div>
    <div data-editable-target=\\"email\\">
        <i>email</i>
    </div>
</div>"
`;

exports[`application 3`] = `
"<div class=\\"container m-3\\">
    <div data-editable-target=\\"name\\">
        <div>Hexlet</div>
    </div>
    <div data-editable-target=\\"email\\">
        <i>email</i>
    </div>
</div>"
`;

exports[`application 4`] = `
"<div class=\\"container m-3\\">
    <div data-editable-target=\\"name\\">
        <div>Hexlet</div>
    </div>
    <div data-editable-target=\\"email\\">
        <form><input type=\\"text\\" name=\\"email\\" value=\\"\\"><input type=\\"submit\\" value=\\"Save\\"></form>
    </div>
</div>"
`;

exports[`application 5`] = `
"<div class=\\"container m-3\\">
    <div data-editable-target=\\"name\\">
        <div>Hexlet</div>
    </div>
    <div data-editable-target=\\"email\\">
        <div>support@hexlet.io</div>
    </div>
</div>"
`;

exports[`application 6`] = `
"<div class=\\"container m-3\\">
    <div data-editable-target=\\"name\\">
        <div><i>name</i></div>
    </div>
    <div data-editable-target=\\"email\\">
        <div>support@hexlet.io</div>
    </div>
</div>"
`;
